app/models/chargeback_rate_detail.rb

```diff
@@ -51,17 +81,16 @@ def friendly_rate
   end
 
   def per_unit_display
-    case per_unit
-    when 'megahertz' then 'MHz'
-    when 'megabytes' then 'MB'
-    when 'gigabytes' then 'GB'
-    when 'kbps' then 'KBps'
-    else per_unit.to_s.capitalize
-    end
+    detail_measure.nil? ? per_unit.to_s.capitalize : detail_measure.measures.key(per_unit)
   end
 
   def rate_type
     # Return parent's rate type
     chargeback_rate.rate_type unless chargeback_rate.nil?
   end
+
+  def detail_measure
+    # Return the measure asociated
+    ChargebackRateDetailMeasure.find_by(:id => chargeback_rate_detail_measure_id)
+  end
```

Comment:
@amaurygonzalez Looks like you forgot to make this change. I'll merge without it and add a PR with the necessary change.